### DISCUSSION TITLE
[Feature] Utility: Correct date from filename on mismatch

### DISCUSSION THREAD

**Discussion by raldone01** [immich-app/immich#12631] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

I have lots of pictures that have at some point lost their correct date but it is still in the filename:

Example filenames:

* `2002_<filename>.<extension>` -> `YYYY`
* `200211_<filename>.<extension>` -> `YYYYMM`
* `20021113_<filename>.<extension>` -> `YYYYMMDD`

A simple utility that checks filenames for various date patterns and compares it to the actual date immich recognizes.
If there is a mismatch it should present the date difference to the user and ask for a solution.

### Possible extension

At a later stage this could possibly be extended by parsing the date for some sort of location like Croatia, Zaton and add a rough geolocation to the image.
It may be useful to look at the full path of the image if it is from an external library.
A tiny llm may be useful for extracting the Location.

### Platform

- [x] Server
- [x] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
#6261
